lib/heap/multi_heap.c

```diff
@@ -90,3 +91,31 @@ void sys_multi_heap_free(struct sys_multi_heap *mheap, void *block)
 		sys_heap_free(heap->heap, block);
 	}
 }
+
+void *sys_multi_heap_realloc(struct sys_multi_heap *mheap, void *cfg, void *block, size_t bytes)
```

Comment:
Devil's advocacy: what if an implementor wants to use the size of an allocation in a multi_heap to discriminate between sub-heaps?  (e.g. things like "big allocations get newly mapped memory from the MMU").  This implementation would disallow that.  Maybe fall back to a top-level sys_multi_heap_alloc() instead?

Not sure which I'd prefer, but it seems like a reasonable question to ask.